tests/ci/run_posix_tests.sh

```diff
@@ -13,6 +13,9 @@ build_and_test
 echo "Testing AWS-LC in release mode."
 build_and_test -DCMAKE_BUILD_TYPE=Release
 
+echo "Testing AWS-LC with Dilithium3 enabled."
+build_and_test -DENABLE_DILITHIUM=ON
```

Comment:
Is there another way? This runs the entire test suite again, duplicating everything except the dilithium tests.

Reply:
That's a fair question, and I agree that adding an additional test suite run is not ideal. Given that we're setting `ENABLE_DILITHIUM=OFF` by default, the testing options I'd identified were:

1. Run all POSIX tests with default `ENABLE_DILITHIUM=OFF`, set `ENABLE_DILITHIUM=ON` as one of the "uncommon build options" in the build-only checks below
2. Run all POSIX tests with non-default `ENABLE_DILITHIUM=ON`,  set `ENABLE_DILITHIUM=OFF` as one of the "uncommon build options" in the build-only checks below
3. Add an independent test run with `ENABLE_DILITHIUM=ON` (i.e. what this PR implements)

Option 1. has the distinct disadvantage of reducing Dilithium coverage to 0, as we'd only be doing compilation checks with the option set. Option 2. would retain Dilithium coverage, but leave the _default_ `ENABLE_DILITHIUM=OFF` build/execution path uncovered by testing, which seems dangerous. That left option 3., where we retain coverage over default and non-default configurations at the cost of (temporarily) increased test runtime.

Option 3. seemed like the safest of the available options.